jgrapht-core/src/main/java/org/jgrapht/generate/NamedGraphGenerator.java

```diff
@@ -48,24 +47,145 @@ public NamedGraphGenerator(VertexFactory<V> vertexFactory)
         vertexMap = new HashMap<>();
     }
 
+
+    // -------------Circulant Graph-----------//
+    /**
+     *
+     * @see #generateCirculantGraph
+     * @param n edge numbers of the graph
+     * @param adjacency proximity that we've to connect the edges between each other
+     * @return Circulant graph
+     */
+    public static Graph<Integer, DefaultEdge> circulantGraph(int n, List<Integer> adjacency)
+    {
+      return circulantGraph(new IntegerVertexFactory(), new ClassBasedEdgeFactory<>(DefaultEdge.class), n, adjacency);
+    }
+
+    /**
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @param vertexFactory a vertex factory
+     * @param edgeFactory an edge factory
+     * @param n edge numbers of the graph
+     * @param adjacency proximity that we've to connect the edges between each other
+     * @return Circulant graph
+     */
+    public static <V,E> Graph<V,E> circulantGraph(VertexFactory<V> vertexFactory, EdgeFactory<V,E> edgeFactory, int n, List<Integer> adjacency)
+    {
+        Graph<V, E> g = new SimpleGraph<>(edgeFactory);
+        new NamedGraphGenerator<V, E>(vertexFactory).generateCirculantGraph(g, n, adjacency);
+        return g;
+    }
+
+    /**
+     * Generates a <a href="http://mathworld.wolfram.com/CirculantGraph.html">Circulan graph</a>. A circulant
+     * graph is a graph of n graph vertices in which the ith graph vertex is adjacent to the (i+j)th and (i-j)th
+     * graph vertices for each j in a list l.
+
+     *
+     * @param targetGraph receives the generated edges and vertices; if this is non-empty on entry,
+     *        the result will be a disconnected graph since generated elements will not be connected
+     *        to existing elements
+     * @param n edge numbers of the graph
+     * @param adjacency proximity that we've to connect the edges between each other
+     */
+    public void generateCirculantGraph(Graph<V, E> targetGraph, int n, List<Integer> adjacency)
+    {
+      vertexMap.clear();
+      for (int i=0;i<n;i++)
+      {
+        for(int j:adjacency)
+        {
+          this.addEdge(targetGraph, i, (i+j)%n);
+          this.addEdge(targetGraph, i, (i-j)%n);
+        }
+      }
+    }
+
+    // -------------Balaban 10 cage-----------//
+    /**
+     * @see #generateBalaban10Cage
+     * @return Balaban 10 Cage graph
+     */
+    public static Graph<Integer, DefaultEdge> balaban10Cage()
+    {
+        return balaban10Cage(new IntegerVertexFactory(), new ClassBasedEdgeFactory<>(DefaultEdge.class));
+    }
+
+    /**
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @param vertexFactory a vertex factory
+     * @param edgeFactory an edge factory
+     * @return Balaban 10 Cage graph
+     */
+    public static <V, E> Graph<V, E> balaban10Cage(VertexFactory<V> vertexFactory, EdgeFactory<V,E> edgeFactory)
```

Comment:
I'm not sure I'm seeing the advantage of these new static methods. Personally it seems to just duplicate the amount of methods without adding anything. If the user wants a graph of a specific vertex/edge type, he's supposed to use the generate...(Graph<V,E> targetGraph) method, in correspondence with the basic graph generator signature.

Reply:
@jkinable It was actually my idea to add these methods. Jairo is an Erasmus student in my department and did this task as an exercise for the data structures course.

Since we decided to add static helpers for the construction of certain small graphs, it makes sense to have also the versions with full generality. Otherwise, one might argue that the static helpers methods are very specific in the choice of vertices and edges.